addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,521 @@
+                return;
+            }
+
+            if (stateReq) {
+                /**
+                 * /api/{username}/{lights}/{id}/{state}
+                 */
+                apiState(id, req, resp);
+            } else if (id != null) {
+                /**
+                 * /api/{username}/{lights}/{id}
+                 */
+                apiLight(id, req, resp);
+            } else if (lightsReq) {
+                /**
+                 * /api/{username}/{lights}
+                 */
+                apiLights(req, resp);
+            } else if (userName != null) {
+                /**
+                 * /api/{username}
+                 */
+                apiDataStore(req, resp);
+            } else {
+                apiServerError(req, resp, HueErrorResponse.NOT_AVAILABLE, "Hue Resource Not Availabale");
+            }
+        }
+
+    }
+
+    /**
+     * Hue API call to set the state of a light
+     *
+     * @param id
+     * @param req
+     * @param resp
+     * @throws IOException
+     */
+    private void apiState(String id, HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        if (!req.getMethod().equals(METHOD_PUT)) {
+            apiServerError(req, resp, HueErrorResponse.METHOD_NOT_AVAILABLE, "Only PUT allowed for this resource");
+            return;
+        }
+        try {
+            Gson gson = new Gson();
+            // will throw exception if not found
+            itemRegistry.getItem(id);
+            HueState state = gson.fromJson(req.getReader(), HueState.class);
+            logger.debug("State " + state);
+            if (state.bri > -1) {
+                eventPublisher.post(ItemEventFactory.createCommandEvent(id,
+                        new DecimalType((int) Math.round(state.bri / 255.0 * 100))));
+            } else {
+                eventPublisher.post(ItemEventFactory.createCommandEvent(id, state.on ? OnOffType.ON : OnOffType.OFF));
+            }
+            PrintWriter out = resp.getWriter();
+            out.write(String.format(STATE_RESP, id, String.valueOf(state.on)));
+            out.close();
+        } catch (ItemNotFoundException e) {
+            logger.warn("Item not found: " + id);
+            apiServerError(req, resp, HueErrorResponse.NOT_AVAILABLE, "The Hue device could not be found");
+        }
+    }
+
+    /**
+     * Hue API call to get the state of a single light
+     *
+     * @param id
+     * @param req
+     * @param resp
+     * @throws IOException
+     */
+    private void apiLight(String id, HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        try {
+            Gson gson = new Gson();
+            Item item = itemRegistry.getItem(id);
+            PrintWriter out = resp.getWriter();
+            out.write(gson.toJson(itemToDevice(item)));
+            out.close();
+        } catch (ItemNotFoundException e) {
+            logger.warn("Item not found: " + id);
```

Comment:
This can simply be an invalid HTTP request - I would not log this as a warning, debug should actually be fine.